backend/prompt_studio/prompt_studio_core_v2/views.py

```diff
@@ -437,13 +437,15 @@ def fetch_contents_ide(self, request: HttpRequest, pk: Any = None) -> Response:
                     file_path += file_name
                 contents = FileManagerHelper.fetch_file_contents(file_system, file_path)
         else:
-            contents = PromptStudioFileHelper.fetch_file_contents(
-                file_name=file_name,
-                org_id=UserSessionUtils.get_organization_id(request),
-                user_id=custom_tool.created_by.user_id,
-                tool_id=str(custom_tool.tool_id),
-            )
-
+            try:
+                contents = PromptStudioFileHelper.fetch_file_contents(
+                    file_name=file_name,
+                    org_id=UserSessionUtils.get_organization_id(request),
+                    user_id=custom_tool.created_by.user_id,
+                    tool_id=str(custom_tool.tool_id),
+                )
+            except FileNotFoundError:
+                raise FileNotFound()
```

Comment:
NIT: We could try to modify this to print the name of the file as well, as you did previously. We might have to override the constructor alone to support passing an argument

Reply:
@chandrasekharan-zipstack This exception class is a legacy one. The proposed change is out of scope for this PR. I do second you for the file name addition in the message. We can note it down and take it up later.